scikits/odes/sundials/cvode.pyx

```diff
@@ -257,15 +264,15 @@ cdef int _jacdense(long int Neq, realtype tt,
         jac_tmp = aux_data.jac_tmp
 
         nv_s2ndarray(yy, yy_tmp)
-    aux_data.jac.evaluate(tt, yy_tmp, jac_tmp)
+    user_flag = aux_data.jac.evaluate(tt, yy_tmp, jac_tmp)
 
     if parallel_implementation:
         raise NotImplemented
     else:
         #we convert the python jac_tmp array to DslMat of sundials
         ndarray2DlsMatd(Jac, jac_tmp)
 
-    return 0
+    return user_flag
```

Comment:
In order not to test too much on None, and leave code as is, perhaps in the generic classes, only update the documentation of them?
For example
``` 
# Jacobian function
cdef class CV_JacRhsFunction:
    cpdef int evaluate(self, DTYPE_t t,
                       np.ndarray[DTYPE_t, ndim=1] y,
                       np.ndarray J) except? -1:
        """
        Returns the Jacobi matrix of the right hand side function, as
            d(rhs)/d y
        (for dense the full matrix, for band only bands). Result has to be
        stored in the variable J, which is preallocated to the corresponding
        size.

        This is a generic class, you should subclass is for the problem specific
        purposes."
        """
        return 0
```
to add that evaluate must return 0 on success, and -1 can be used to interrupt the solver.